mitodl/mitxonline
# Department REST API

# What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1876

# Description (What does it do?)
Adds a REST API endpoint for retrieving departments in MITx Online.  The endpoint is http://mitxonline.odl.local:8013/api/departments.  The data returned from the endpoint is:
```
[
    {
        "name": "Science",
        "courses": 2,
        "programs": 0
    },
    {
        "name": "Math",
        "courses": 1,
        "programs": 0
    },
    {
        "name": "test",
        "courses": 1,
        "programs": 1
    }
]
```

Calls to this endpoint are added from the catalog page in order to display all departments for courses or programs upon initial load of the page.

Departments are only displayed based on the following:
1. If the current tab is "courses", the department must have 1 or more courses associated with it.
2. If the current tab is "programs, the department must have 1 or more programs associated with it.

# How can this be tested?

1. Create 13 courses and course runs which are expected to be displayed in the catalog.
2. Add a unique department to the 13th course.
3. Load the catalog page. Verify that the unique department associated with the 13th course is shown even if the 13th course has not yet been displayed on the catalog page.

# Additional Context
There is the possibility that a department is associated with 1 or more courses or programs that will not be visible on the catalog page due to the catalogs business logic for filtering courses and programs.  The department will still be shown.
